FB Playable Ad documentation (#241)
playcanvas/developer.playcanvas.com: content/en/user-manual/publishing/playable-ads/fb-playable-ads.md

---
title: Facebook Playable Ad
template: usermanual-page.tmpl.html
position: 3
---

PlayCanvas supports the [Facebook Playable Ad][1] formats and requirements via an [official external tool on GitHub][2].

<video loop controls autoplay muted src="/videos/fb-playable-ad-bitmoji-creator.mp4" width="200"> </video> <video loop controls autoplay muted src="/videos/fb-playable-ad-cube-jump.mp4" width="200"></video> <video loop controls autoplay muted src="/videos/fb-playable-ad-flappy-bird.mp4" width="200"></video>

> The tool can create both the single 2MB (uncompressed) HTML file and the 5MB (uncompressed) ZIP formats via the configuration options. Full specifications for Facebook Playable Ads can be found on their [help centre][3].

Reviewer: Should this say `5MB (compressed)`?

Author: I'm under the impression that is uncompressed (the docs are really not clear) https://www.facebook.com/business/help/412951382532338?ref=FBB_PlayableAds

```
The total size of uploaded files may not exceed 5 MB.
The number of files inside the bundle may not exceed 100.
The size of index.html file should not exceed 2 MB.
```

> There are some limitations to be aware of with the tool which can be found on the documentation from [GitHub][2].

Reviewer: ```suggestion
There are some limitations to be aware of with the tool which can be found in the documentation from [GitHub][2].
```

## Example project

> [Cube Jump project][5] is ready to exported for the Facebook Playable Ad format and the expected [HTML output can be found here][6].

Reviewer: ```suggestion
The [Cube Jump project][5] is ready to be exported for the Facebook Playable Ad format and the expected [HTML output can be found here][6].
```

<iframe src="https://playcanv.as/e/p/Hywjl9Bh/"></iframe>

## File size tips

As there is a strict file size limit, you will have to plan and budget the usage of assets for the ad.

The minified PlayCanvas Engine code is **~1.2MB** uncompressed and due to the need to encode the asset files into Base64 strings, it adds **~30%** to the size of each asset file.

This means that for a single HTML format, this leaves ~500KB for assets before they are encoded into Base64 strings.

Try to keep images as small as possible in dimensions and use tools like [TinyPNG][4] to reduce file size even further.

## How to export

> Follow the [setup steps][7] from the read me in the GitHub repo.

Reviewer: ```suggestion
Follow the [setup steps][7] from the readme in the GitHub repo.
```
